libgadget/fof.c

```diff
@@ -796,7 +796,7 @@ fof_compile_catalogue(struct Group * group, double BoxSize, int BlackHoleInfo, M
     fof_finish_group_properties(Group, BoxSize);
 
     int64_t TotNids;
-    MPI_Allreduce(&Ngroups, &TotNgroups, 1, MPI_UINT64, MPI_SUM, Comm);
+    sumup_large_ints(1, &Ngroups, &TotNgroups);
```

Comment:
It's cleaner to use TotNgroups = Ngroups and MPI_IN_PLACE.
Most use cases of sumup_large_ints can be replaced by this.

Reply:
Ah yes - that way the right communicator is used. Thanks.